main.tf

```diff
@@ -160,22 +160,27 @@
   * ```
   */
 
+locals {
+  default_tags = {
+    envname = "${var.envname}"
+    envtype = "${var.envtype}"
+  }
+}
+
 // DB Subnet Group creation
 resource "aws_db_subnet_group" "main" {
   count       = "${var.enabled ? 1 : 0}"
   name        = "${var.name}"
   description = "Group of DB subnets"
   subnet_ids  = ["${var.subnets}"]
 
-  tags {
-    envname = "${var.envname}"
-    envtype = "${var.envtype}"
-  }
+  tags = "${merge("${local.default_tags}","${var.tags}")}"
```

Comment:
This should do the same thing with less mess:
```
tags = "${merge(local.default_tags, var.tags)}"
```

Reply:
Great suggestion, cleaner merge syntax. I got literal happy. See revision.